### DISCUSSION TITLE
Undelegation bug after Indexer Unregister

### DISCUSSION THREAD

**Discussion by bobyD** [subquery/subql#1055] [Frontier TestNet - Bug Reporting]
After the indexer will be unregistered from the network and receive `INACTIVE` status on `My profile -> Delegating` page, there is no way to undelegate tokens from this indexer. Explorer shows the following tooltip: `This indexer cannot be undelegated from until they collect all the early era's rewards. Please contact the indexer to resolve the issue`

First, I saw this problem with the another indexer `0xAF2c0A4495E66FB1f27F553Dc8376bBc4560f839` after it was unregistered from the network. Then I tried it with my own indexer `0x848488C3daa5d31E43E8d02af6EaE61f8Fe41d6d` and received the same bug. Initially `Undelegate` button was available in the era when the `Unregister` action was proceeded successfully, without any errors. But after the current era finished, the button became unavailable with the error tooltip.

Attached few screenshots with my delegation to another indexer and with the delegation to my indexer from a different address. Both cases happened after indexers were unregistered from the network.
<img width="1430" alt="Снимок экрана 2022-05-22 в 11 25 13" src="https://user-images.githubusercontent.com/69699206/169710349-b5be832e-38f2-4c94-9804-034ea558c15e.png">
<img width="1440" alt="Снимок экрана 2022-05-22 в 21 18 34" src="https://user-images.githubusercontent.com/69699206/169710361-d40530e3-22f3-40c1-8210-5d77df9da52f.png">

Before I have unregistered my indexer, I checked the logs to see that collected and distributed rewards have `SUCCEED` status in the current era

_QuantNode team_

**Comment by mzxyz** [comment]
Thanks for report this issue, we will better handle the flow for indexer unregister.
